tests/python/pants_test/backend/python/tasks/test_python_distribution_integration.py

```diff
@@ -70,23 +75,51 @@ def test_with_conflicting_transitive_deps(self):
     self.assertIn('fasthello', pants_run.stderr_data)
 
   def test_pants_binary_dep_isolation_with_multiple_targets(self):
-    command=['binary', '{}:main_with_no_conflict'.format(self.fasthello_install_requires),
-             '{}:main_with_no_pycountry'.format(self.fasthello_install_requires)]
-    pants_run = self.run_pants(command=command)
-    self.assert_success(pants_run)
-    # Check that the pex was built.
     pex1 = os.path.join(get_buildroot(), 'dist', 'main_with_no_conflict.pex')
-    self.assertTrue(os.path.isfile(pex1))
     pex2 = os.path.join(get_buildroot(), 'dist', 'main_with_no_pycountry.pex')
-    self.assertTrue(os.path.isfile(pex2))
-    # Check that the pex 1 runs.
-    output = subprocess.check_output(pex1)
-    self.assertIn('Super hello', output)
-    # Check that the pex 2 fails due to no python_dists leaked into it.
     try:
-      output = subprocess.check_output(pex2)
-    except subprocess.CalledProcessError as e:
-      self.assertNotEquals(0, e.returncode)
-    # Cleanup
-    os.remove(pex1)
-    os.remove(pex2)
+      command=['binary', '{}:main_with_no_conflict'.format(self.fasthello_install_requires),
+               '{}:main_with_no_pycountry'.format(self.fasthello_install_requires)]
+      pants_run = self.run_pants(command=command)
+      self.assert_success(pants_run)
+      # Check that the pex was built.
+      self.assertTrue(os.path.isfile(pex1))
+      self.assertTrue(os.path.isfile(pex2))
+      # Check that the pex 1 runs.
+      output = subprocess.check_output(pex1)
+      self.assertIn('Super hello', output)
+      # Check that the pex 2 fails due to no python_dists leaked into it.
+      try:
+        output = subprocess.check_output(pex2)
+      except subprocess.CalledProcessError as e:
+        self.assertNotEquals(0, e.returncode)
+    finally:
+      # Cleanup
+      if os.path.exists(pex1):
+        os.remove(pex1)
+      if os.path.exists(pex2):
+        os.remove(pex2)
+
+  def test_pants_resolves_local_dists_for_current_platform_only(self):
+    # Test that pants will override pants.ini platforms config when building
+    # or running a target that depends on native (c or cpp) sources.
+    pex = os.path.join(get_buildroot(), 'dist', 'main.pex')
+    pants_ini_config = {'python-setup': {'platforms': ['current', 'linux-x86_64']}}
```

Comment:
It's not clear to me what this is testing. Is this testing that pants ignores the python-setup platforms (it seems like it), or is Pants supposed to check against these platforms for compatibility with the host platform? Why specifically `'linux-x86_64'` and not also `'macosx-10.8-intel'`?